tiatoolbox/models/architecture/nuclick.py

```diff
@@ -0,0 +1,640 @@
+        activation: str = "relu",
+        do_batchnorm: bool = True,
+    ):
+
+        super().__init__()
+        if isinstance(kernel_size, int):
+            kernel_size = (kernel_size, kernel_size)
+        if isinstance(strides, int):
+            strides = (strides, strides)
+
+        self.conv_bn_relu = self.get_block(
+            num_input_channels,
+            num_output_channels,
+            kernel_size,
+            strides,
+            use_bias,
+            dilation_rate,
+            activation,
+            do_batchnorm,
+        )
+
+    def forward(self, input_tensor: torch.Tensor) -> torch.Tensor:
+        """Logic for using layers defined in init.
+
+        This method defines how layers are used in forward operation.
+
+        Args:
+            input_tensor (torch.Tensor):
+                    Input, the tensor is of the shape NCHW.
+        Returns:
+            output (torch.Tensor):
+                        The inference output.
+
+
+        """
+        return self.conv_bn_relu(input_tensor)
+
+    @staticmethod
+    def get_block(
+        in_channels,
+        out_channels,
+        kernel_size,
+        strides,
+        use_bias,
+        dilation_rate,
+        activation,
+        do_batchnorm,
+    ):
+        """Function to acquire a convolutional block.
+
+        Args:
+            in_channels (int):
+                Number of channels in input.
+            out_channels (int):
+                Number of channels in output.
+            kernel_size (list):
+                Size of the kernel in the acquired convolution block.
+            strides (int):
+                Size of stride in the convolution layer.
+            use_bias (bool):
+                Whether to use bias in the convolution layer.
+            dilation_rates (list):
+                Dilation rate for each convolution layer.
+            activation (str):
+                Name of the activation function to use.
+            do_batchnorm (bool):
+                Whether to do batch normalization after the convolution layer.
+
+        Returns:
+            torch.nn.Sequential: a pytorch layer
+
+        """
+        conv1 = nn.Conv2d(
+            in_channels=in_channels,
+            out_channels=out_channels,
+            kernel_size=kernel_size,
+            stride=strides,
+            dilation=dilation_rate,
+            bias=use_bias,
+            padding="same",
+            padding_mode="zeros",
+        )
+
+        torch.nn.init.xavier_uniform_(conv1.weight)
+
+        layers = [conv1]
+
+        if do_batchnorm:
+            layers.append(nn.BatchNorm2d(num_features=out_channels, eps=1.001e-5))
+
+        if activation == "relu":
+            layers.append(nn.ReLU())
+
+        return nn.Sequential(*layers)
+
+
+class MultiscaleConvBlock(nn.Module):
+    """Defines Multiscale convolution block.
+
+    Args:
+        num_input_channels (int):
+            Number of channels in input.
+        num_output_channels (int):
```

Comment:
```suggestion
        num_class (int):
```

Shouldn't we call it num_class as this is the number of output classes?
If you think we should keep it num_output_channels then please refactor this in other models as well.

Reply:
Thanks for the suggestion but I think it's better to keep it as it is to be in line with other networks (UNet) implementation in the toolbox.

Reply:
The number of output classes may be inferred differently with different networks. Some might use a FC layer with the same number of nodes as "number of output classes" to classify/detect/segment the objects in the image, other methods might use a number of output channels to identify a single class (instance segmentation where we have marker and mask), and some methods that use only one channel to detect/segment different "number of output classes".

My point is that these are valid and invalid. I agree that it should be consistent through the toolbox but might be impossible due to the nature of different models. Anyway, I tried to keep this network architecture in line with UNet implementation as much as possible.